store/views.py

```diff
@@ -77,6 +107,52 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    response_data = {
+        'message': "failure"
+    }
+    data = request.POST
+    copybook_id = data.get('bid')
+    if copybook_id is not None:
+        copybook = BookCopy.objects.get(id=copybook_id)
+        copybook.borrow_date = None
+        copybook.borrower_id = None
+        copybook.status = False
+        copybook.save()
+        response_data['message']="success"
+    return JsonResponse(response_data)
+
+@csrf_exempt
+@login_required
+def rateBookView(request):
+    response_data = {
+        'message': "failure"
+    }
+    prating = '-1'
+    userid=request.user.id
+    data = request.POST
+    rating =data.get('rating')
+    bookid = data.get('bid')
+    print(rating)
+    if Rating.objects.filter(user_id=userid).filter(book_id=bookid).count() == 1:
+        prating = Rating.objects.filter(user_id=userid).filter(book_id=bookid).first().rating
+        print('rating')
+    if prating == '-1':
+        rate = Rating.objects.create(book_id=bookid,user_id=userid,rating=rating)
+        response_data['message'] = "success"
+    else :
+        rate=Rating.objects.filter(user_id=userid).filter(book_id=bookid).first()
+        rate.rating = rating
```

Comment:
You've not put a backend validation on the rating, so the user can simply edit the JS code you've written in the template and easily put invalid values of rating.